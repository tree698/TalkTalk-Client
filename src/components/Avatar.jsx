import React from 'react';

export default function Avatar({ photo, username, styleForAvatar }) {
  return (
    <div>
      {!!photo ? (
        <img
          className={styleForAvatar}
          src={`${process.env.REACT_APP_BASE_URL}/uploaded_images/${photo}`}
        />
      ) : (
        <div className="flex justify-center items-center w-8 h-8 bg-brand text-white text-xl rounded-full mr-2">
          {username.charAt(0)}
        </div>
      )}
    </div>
  );
}
