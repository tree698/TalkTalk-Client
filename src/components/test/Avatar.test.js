import { render, screen } from '@testing-library/react';
import { Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { withRouter } from '../../test/utils';
import Avatar from '../Avatar';

describe('Avatar', () => {
  it('renders correctly with photo', () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<Avatar photo={true} />} />)
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('display photo', () => {
    render(withRouter(<Route path="/" element={<Avatar photo={'photo'} />} />));

    const image = screen.getByRole('img');
    expect(image.src).toBe('http://localhost/photo');
  });

  it('display username', () => {
    render(
      withRouter(<Route path="/" element={<Avatar username={'username'} />} />)
    );

    expect(screen.getByText('u')).toBeInTheDocument();
  });
});
