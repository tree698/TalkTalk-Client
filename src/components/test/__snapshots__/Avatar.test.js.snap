// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly with photo 1`] = `
<div>
  <img
    src={true}
  />
</div>
`;
